File: boj/DynamicProgramming/BOJ_1149.py
# 백준 1463 RGB거리
# 21.02.22

import sys
from functools import cmp_to_key
from collections import deque
from itertools import combinations
from itertools import permutations
sys.setrecursionlimit(10**6)

n = int(input())

d = []
for _ in range(n):
    d.append(list(map(int, sys.stdin.readline().split())))

for i in range(1, n):
    d[i][0] += min(d[i-1][1], d[i-1][2])
    d[i][1] += min(d[i-1][0], d[i-1][2])
    d[i][2] += min(d[i-1][0], d[i-1][1])

print(min(d[n-1]))

File: boj/DynamicProgramming/BOJ_1149.swift
// 백준 1149 RGB거리
// 21.02.24

import Foundation

let n = Int(readLine()!)!

var rgb = readLine()!.split(separator: " ").map { Int(String($0))! }

for _ in stride(from: 1, to: n, by: 1) {
  var next = readLine()!.split(separator: " ").map { Int(String($0))! }
  next[0] += min(rgb[1], rgb[2])
  next[1] += min(rgb[0], rgb[2])
  next[2] += min(rgb[0], rgb[1])
  
  rgb[0] = next[0]; rgb[1] = next[1]; rgb[2] = next[2]
}

print(min(rgb[0], min(rgb[1], rgb[2])))
